### ISSUE TITLE
Seems socket.Server.Emit not work?

### ISSUE THREAD

**Issue by funte** [zishang520/socket.io#71]
Expected:
The postman client should get "server hello..." message but not.
<img width="767" alt="image" src="https://github.com/user-attachments/assets/3f03497a-0d9c-48f1-94be-35db7b1bc6aa">

Reproduce code:
```go
package main

import (
	"net/http"

	"github.com/go-chi/chi/v5"
	"github.com/go-chi/chi/v5/middleware"
	"github.com/zishang520/socket.io/v2/socket"
)

func main() {
	var r = chi.NewRouter()
	r.Use(middleware.Heartbeat("/ping"))
	var server = socket.NewServer(nil, nil)
	server.On("connection", func(clients ...any) {
		var client = clients[0].(*socket.Socket)
		client.Emit("welcome", "client hello...")
		server.Emit("welcome", "server hello...")
		server.Of("/", nil).Emit("welcome", "root room hello...")
	})
	r.Handle("/socket.io/", server.ServeHandler(nil))
	http.ListenAndServe("localhost:8080", r)
}

```

**Comment by zishang520** [comment]
Waiting for my survey results.

**Comment by zishang520** [comment]
This issue has been fixed in v2.3.5:

![image](https://github.com/user-attachments/assets/32653812-5661-447b-a3b7-9404c69b4b42)
![image](https://github.com/user-attachments/assets/0734593d-de7f-46d6-b033-89d56fd39da9)
